lib/CL/devices/remote/communication.c

```diff
@@ -2416,7 +2422,7 @@ pocl_network_build_program (remote_device_data_t *ddata, const void *payload,
   POCL_MSG_PRINT_REMOTE ("METADATA SIZE: %zu\n", (size_t)metadata_size);
 
   /* read program binaries */
-  if (!is_binary && !is_builtin)
+  if (!is_binary && !is_builtin && !is_spirv)
```

Comment:
Don't we want to read the binaries for SPIR-V compilation ?

Reply:
Good catch!